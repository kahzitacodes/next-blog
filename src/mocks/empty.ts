import { EmptyProps } from '@/components/empty'

export const emptyMock: EmptyProps = {
  title: 'Oops!',
  description: 'Page not found'
}
