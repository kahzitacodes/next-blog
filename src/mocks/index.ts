export * from './navigation'
export * from './profile'
export * from './social-links'
export * from './post'
export * from './pre'
export * from './pagination'
