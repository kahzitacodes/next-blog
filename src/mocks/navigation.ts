import { NavItem } from '@/types'

export const navigationMock: NavItem[] = [
  {
    title: 'Home',
    href: '/'
  },
  {
    title: 'About',
    href: '/about'
  },
  {
    title: 'Contact',
    href: '/contact'
  }
]
