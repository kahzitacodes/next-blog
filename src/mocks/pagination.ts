export const paginationMock = {
  currentPage: 1,
  totalPages: 10
}
