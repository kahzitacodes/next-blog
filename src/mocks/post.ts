import { Post } from '@/types'

export const postMock: Post = {
  $frontMatter: {
    date: '2025-04-04',
    description: 'Description',
    image: '/assets/images/image.png',
    title: 'Post title',
    tags: ['react']
  },
  $readingTime: 3,
  body: {
    code: '',
    raw: ''
  },
  id: '1',
  slug: 'post-3'
}
