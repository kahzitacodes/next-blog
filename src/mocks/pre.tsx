import { PreProps } from '@/components/pre'

export const preMock: PreProps = {
  children: <h1>Title</h1>
}
