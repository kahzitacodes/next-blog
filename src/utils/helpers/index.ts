export * from './slugify'
export * from './format-date'
export * from './format-post'
export * from './pagination'
