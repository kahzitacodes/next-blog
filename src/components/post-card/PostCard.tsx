'use client'

import { formatDate } from '@/utils/helpers'
import { Tag } from '../tag'

import { PropTypes } from './PostCard.types'
import * as S from './PostCard.styles'

export const PostCard: React.FC<PropTypes> = (props) => {
  const {
    $readingTime,
    slug,
    $frontMatter: { description, image, title, date, tags }
  } = props

  const postLink = `/articles/${slug}`

  return (
    <S.PostContainer>
      <S.PostImageContainer href={postLink}>
        <S.PostImage src={image} fill alt={title} priority />
      </S.PostImageContainer>

      {tags ? (
        <S.TagsWrap>
          {tags.map((tag) => (
            <Tag key={tag} label={tag} />
          ))}
        </S.TagsWrap>
      ) : null}

      <S.Details>
        {formatDate(date)} â€¢ {$readingTime.toString()} minuto
        {$readingTime <= 1 ? '' : 's'} de leitura
      </S.Details>

      <S.Content href={postLink} className="">
        <S.Title>{title}</S.Title>
        <S.Summary>{description}</S.Summary>
      </S.Content>
    </S.PostContainer>
  )
}
