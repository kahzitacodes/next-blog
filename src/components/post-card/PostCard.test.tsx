import { render, screen, within } from '@/utils/tests/custom-render'
import { PostCard } from './PostCard'
import { postMock } from '@/mocks'
import { formatDate } from '@/utils/helpers'

describe('<Post />', () => {
  it('should render successfully', () => {
    const { container } = render(<PostCard {...postMock} />)
    expect(screen.getByAltText(postMock.$frontMatter.title)).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: postMock.$frontMatter.title })
    ).toBeInTheDocument()
    expect(screen.getByText(postMock.$frontMatter.description))

    expect(container).toMatchSnapshot()
  })

  it('should render tags correctly', () => {
    render(<PostCard {...postMock} />)

    const tagList = screen.getByRole('list', {
      name: /tags/i
    })
    const { getAllByRole } = within(tagList)
    const tags = getAllByRole('listitem')

    expect(tags.length).toBe(2)
  })

  it('should render the date and reading time correctly', () => {
    render(<PostCard {...postMock} />)

    const formattedDate = formatDate(postMock.$frontMatter.date)
    const readingTime = postMock.$readingTime
    const minuteText = readingTime <= 1 ? 'minuto' : 'minutos'
    const fullText = `${formattedDate} â€¢ ${readingTime.toString()} ${minuteText} de leitura`

    expect(screen.getByText(fullText)).toBeInTheDocument()
  })
})
