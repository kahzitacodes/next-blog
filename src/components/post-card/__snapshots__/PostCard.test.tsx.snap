// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> should render successfully 1`] = `
<div>
  <article
    class="flex flex-col gap-5"
  >
    <a
      class="relative h-80 w-full rounded-lg overflow-hidden"
      href="/posts/post-3"
    >
      <img
        alt="Post title"
        class="object-cover object-center transition-all hover:scale-105"
        data-nimg="fill"
        decoding="async"
        fetchpriority="high"
        sizes="100vw"
        src="/_next/image?url=%2Fassets%2Fimages%2Fimage.png&w=3840&q=75"
        srcset="/_next/image?url=%2Fassets%2Fimages%2Fimage.png&w=640&q=75 640w, /_next/image?url=%2Fassets%2Fimages%2Fimage.png&w=750&q=75 750w, /_next/image?url=%2Fassets%2Fimages%2Fimage.png&w=828&q=75 828w, /_next/image?url=%2Fassets%2Fimages%2Fimage.png&w=1080&q=75 1080w, /_next/image?url=%2Fassets%2Fimages%2Fimage.png&w=1200&q=75 1200w, /_next/image?url=%2Fassets%2Fimages%2Fimage.png&w=1920&q=75 1920w, /_next/image?url=%2Fassets%2Fimages%2Fimage.png&w=2048&q=75 2048w, /_next/image?url=%2Fassets%2Fimages%2Fimage.png&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </a>
    <div
      class="flex flex-col"
    >
      <ul
        aria-label="tags"
        class="mb-4 flex flex-wrap gap-2"
      >
        <li>
          <span
            class="flex flex-col rounded-full px-2 py-0.5 text-sm font-medium cursor-default transition-all h-fit bg-gray-300 text-gray-700"
            variant="default"
          >
            react
          </span>
        </li>
        <li>
          <span
            class="flex flex-col rounded-full px-2 py-0.5 text-sm font-medium cursor-default transition-all h-fit bg-gray-300 text-gray-700"
            variant="default"
          >
            typescript
          </span>
        </li>
      </ul>
      <div
        class="text-sm text-gray-400 mb-2"
      >
        4 de abril de 2025
         â€¢ 
        3
         minuto
        s
         de leitura
      </div>
      <a
        class="cursor-pointer [&>h3]:hover:text-link [&>p]:hover:text-white"
        href="/posts/post-3"
      >
        <h3
          class="mt-2 max-w-md text-ellipsis text-2xl font-medium text-gray-50 transition-colors"
        >
          Post title
        </h3>
        <p
          class="mt-3 text-gray-400"
        >
          Description
        </p>
      </a>
    </div>
  </article>
</div>
`;
